nibabel/spatialimages.py

```diff
@@ -866,6 +867,22 @@ def from_image(klass, img):
                      klass.header_class.from_header(img.header),
                      extra=img.extra.copy())
 
+    @classmethod
+    def is_image(klass, filename, sniff=None):
+        ftypes = dict(klass.files_types)
+        froot, ext, trailing = splitext_addext(filename, ('.gz', '.bz2'))
+        lext = ext.lower()
+
+        if lext not in ftypes.values():
+            return False, sniff
+
+        fname = froot + ftypes['header'] if 'header' in ftypes else filename
+        if not sniff:
+            with BinOpener(fname, 'rb') as fobj:
+                sniff = fobj.read(1024)
```

Comment:
Perhaps allow header classes to define a `sniff_size`. In that case, only sniff if `header_class.sniff_size` is defined, `header_class.is_header` is defined. 

That would also simplify the `loadsave` logic (no special-casing for len(possible_classes) == 0) and avoid a potential problem if `header_class.is_header` is not defined.